src/allmydata/test/test_node.py

```diff
@@ -584,15 +606,30 @@ def test_listen_on_zero(self):
             str(ctx.exception),
         )
 
-    def test_multiple_ports(self):
+    # Randomly allocate a couple distinct port numbers to try out.  The test
+    # never actually binds these port numbers so we don't care if they're "in
+    # use" on the system or not.  We just want a couple distinct values we can
+    # check expected results against.
+    @given(ports=sets(elements=port_numbers(), min_size=2, max_size=2))
+    # A few examples should satisfy us here.  The logic for dealing with these
+    # numbers is hardly complex.  On the flip side, all of the setup work we
+    # do to get a node we can test against is pretty expensive.
+    @settings(max_examples=10)
+    def test_multiple_ports(self, ports):
+        """
+        When there are multiple listen addresses suggested by the ``tub.port`` and
+        ``tub.location`` configuration, the node's *main* port listens on all
+        of them.
+        """
         basedir = self.mktemp()
-        create_node_dir(basedir, "testing")
-        port1 = iputil.allocate_tcp_port()
-        port2 = iputil.allocate_tcp_port()
+        config_fname = os.path.join(basedir, "tahoe.cfg")
+        os.mkdir(basedir)
+        os.mkdir(os.path.join(basedir, "private"))
+        port1, port2 = iter(ports)
         port = ("tcp:%d:interface=127.0.0.1,tcp:%d:interface=127.0.0.1" %
```

Comment:
Probably should be if we aren't.  I try to use it in new code but I'm reluctant to convert it drive-by for other changes for a number of reasons, but particularly because Tahoe-LAFS seems to have a bunch of opinions about how to deal with encoding and file names and I don't know if those opinions align with FilePath.  Not that that's relevant here since this code is just os.path and such.